paper-typeahead-input.html

```diff
@@ -449,6 +450,14 @@
         // }
       }
     },
+    _candidatesUpdated:function(candidates){
+      if(this.inputValue != undefined){
+        var val = this.inputValue.trim();
+        if(candidates.length > 0 && candidates[0].trim() !== val){
```

Comment:
Because the observer function is called when localCandidates variables change. We need to think as many cases as possible. The implementation shouldn't rely on your specific use of this function. Otherwise we might face bugs if this case happens. Theoretically it could happen somehow.